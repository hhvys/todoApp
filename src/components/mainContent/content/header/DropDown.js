import React from 'react';
import SortCreation from "../../../logos/SortCreation";
import Star from '../../../logos/Star';
import Sort from '../../../logos/Sort';
import {SORT_BY} from '../../../../actions/actionTypes';
import '../../../logos/logos.css';

const DropDown = ({onClick, className, changeSort}) => {
	return (
		<div onClick={onClick} className={className}>
			<div className="options d-flex align-items-center" onClick={() => changeSort(SORT_BY.SORT_ALPHA)}>
				<div className="symbol d-flex align-items-center">
					<Sort style={{fill: 'white'}}/>
				</div>
				<div className="symbol-content">
					Sort Alphabetically
				</div>
			</div>

			<div className="options d-flex align-items-center" onClick={() => changeSort(SORT_BY.SORT_CREATION)}>
				<div className="symbol d-flex align-items-center">
					<SortCreation style={{fill: 'white'}}/>
				</div>
				<div className="symbol-content">
					Sort by Creation Date
				</div>
			</div>

			<div className="options d-flex align-items-center" onClick={() => changeSort(SORT_BY.SORT_PRIORITY)}>
				<div className="symbol d-flex align-items-center">
					<Star style={{fill: 'white'}}/>
				</div>
				<div className="symbol-content">
					Sort by Priority
				</div>
			</div>

		</div>
	);
};

export default DropDown;