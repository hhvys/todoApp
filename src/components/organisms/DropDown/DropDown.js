import React from 'react';
import styles from './DropDown.mod.scss';
import VerticalTab from "../../molecules/VerticalTab/VerticalTab";
import {SORT, STAR, SORT_CREATION} from "../../atoms/logos/constants";
import {SORT_BY} from "../../../actions/actionTypes";

class DropDown extends React.Component {


	render() {
		const {
			className,
			onClick,
			style,
			onTabClick,
			...props
		} = this.props;
		return (
			<div className={className} onClick={onClick} {...props} style={{...style}}>
				<VerticalTab
					className={styles.tab}
					onClick={() => onTabClick(SORT_BY.SORT_ALPHA)}
					headerSymbol={SORT}
					mainContent={"Sort Alphabetically"}
					headerClass={styles.headerLogo}
				/>
				<VerticalTab
					className={styles.tab}
					onClick={() => onTabClick(SORT_BY.SORT_CREATION)}
					headerSymbol={SORT_CREATION}
					mainContent={"Sort by Creation Date"}
					headerClass={styles.headerLogo}
				/>
				<VerticalTab
					className={styles.tab}
					onClick={() => onTabClick(SORT_BY.SORT_PRIORITY)}
					headerSymbol={STAR}
					mainContent={"Sort by Priority"}
					headerClass={styles.headerLogo}
				/>
			</div>
		);
	}
}

export default DropDown;
