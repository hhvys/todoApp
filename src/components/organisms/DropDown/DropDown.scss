.tab {
  height: 31px;
  color: white;
}

.tab:hover {
  background-color: #769475;
}

.headerLogo {
  fill: white;
  width: 30px;
}
