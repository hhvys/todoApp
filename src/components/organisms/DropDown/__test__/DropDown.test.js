import React from 'react';
import DropDown from '../DropDown';

describe('DropDown', () => {
	it('should render', () => {
		const props = {
			className: "className"
		};
		expect(toJSON(shallow(<DropDown {...props}/>))).toMatchSnapshot();
	});

	it('should pass onTabClick as onClick prop to RowComponents', () => {
		const props = {
			onTabClick: jest.fn()
		};
		expect(props.onTabClick.mock.calls.length).toBe(1);
	});

	it('should pass onClick', () => {
		const props = {
			onClick: jest.fn()
		};
		expect(toJSON(shallow(<DropDown {...props}/>))).toMatchSnapshot();
	});

});
