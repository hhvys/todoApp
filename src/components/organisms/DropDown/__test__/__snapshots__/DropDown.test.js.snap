// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown should pass onClick 1`] = `
<div
  onClick={[Function]}
>
  <RowComponent
    className="tab"
    headerClass="headerLogo"
    headerSymbol="SORT"
    mainContent="Sort Alphabetically"
    onClick={[Function]}
  />
  <RowComponent
    className="tab"
    headerClass="headerLogo"
    headerSymbol="SORT_CREATION"
    mainContent="Sort by Creation Date"
    onClick={[Function]}
  />
  <RowComponent
    className="tab"
    headerClass="headerLogo"
    headerSymbol="STAR"
    mainContent="Sort by Priority"
    onClick={[Function]}
  />
</div>
`;

exports[`DropDown should render 1`] = `
<div
  className="className"
>
  <RowComponent
    className="tab"
    headerClass="headerLogo"
    headerSymbol="SORT"
    mainContent="Sort Alphabetically"
    onClick={[Function]}
  />
  <RowComponent
    className="tab"
    headerClass="headerLogo"
    headerSymbol="SORT_CREATION"
    mainContent="Sort by Creation Date"
    onClick={[Function]}
  />
  <RowComponent
    className="tab"
    headerClass="headerLogo"
    headerSymbol="STAR"
    mainContent="Sort by Priority"
    onClick={[Function]}
  />
</div>
`;
